### PR TITLE
Don't send empty kms_arn in glue_security_configuration if mode is DISABLED

### LINKED ISSUE #13620: aws_glue_security_configuration send empty kms_arn in DISABLED configs
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

v0.12.26

### Affected Resource(s)

* aws_glue_security_configuration

### Terraform Configuration Files

```hcl
resource "aws_glue_security_configuration" "s3_encryption" {
  name = "xaldigital-test"

  encryption_configuration {
    cloudwatch_encryption {
      cloudwatch_encryption_mode = "DISABLED"
    }

    job_bookmarks_encryption {
      job_bookmarks_encryption_mode = "DISABLED"
    }

    s3_encryption {
      kms_key_arn        = aws_kms_key.key.arn
      s3_encryption_mode = "SSE-KMS"
    }
  }
}
```

### Debug Output

https://gist.github.com/AndresCidoncha/70cac98df4d5fd5c5e0c5b91f92c9103

### Expected Behavior

Terraform must omit the kms_arn fields in the API call, so the security_configuration should look like this:

```json
"CloudWatchEncryption":{"CloudWatchEncryptionMode":"DISABLED"},
"JobBookmarksEncryption":{"JobBookmarksEncryptionMode":"DISABLED"}
```

image

### Actual Behavior

Instead, empty values for kms_arn are sent in the API call, so AWS creates the security_config with empty values for key_arn:

```json
"CloudWatchEncryption":{"CloudWatchEncryptionMode":"DISABLED","KmsKeyArn":""},"JobBookmarksEncryption":{"JobBookmarksEncryptionMode":"DISABLED","KmsKeyArn":""}
```

image

Running an ETL with this security_config causes this error:

image

### Steps to Reproduce

1. `terraform apply`

### References

https://github.com/terraform-providers/terraform-provider-aws/pull/13618/

### PR DESCRIPTION
**Pull request by AndresCidoncha**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #13620 

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSGlueSecurityConfiguration_Basic'

...
```

### CODE REVIEW

**Comment by DrFaust92** [comment]
Hey @AndresCidoncha, can you also add an acceptance test to verify the use case?

**Comment by AndresCidoncha** [comment]
> Hey @AndresCidoncha, can you also add an acceptance test to verify the use case?

Hey @DrFaust92 , I wanted to do a new test but I found a problem: The empty configuration is already tested in `TestAccAWSGlueSecurityConfiguration_Basic` but the `TestCheckResourceAttr` method only accepts a string for the comparation value and it's currently using an empty string, so I can't check if the assigned value is nil instead of a empty string.

There's a way to validate the fields sent in the request to AWS?

File: aws/resource_aws_glue_security_configuration.go [hunk 1]
```diff
@@ -214,7 +214,9 @@ func expandGlueCloudWatchEncryption(l []interface{}) *glue.CloudWatchEncryption
 	}
 
 	if v, ok := m["kms_key_arn"]; ok {
```
Comment (DrFaust92) [inline aws/resource_aws_glue_security_configuration.go:216]: lets remove the extra if by doing this:
```
if v, ok := m["kms_key_arn"]; ok  && v.(string) != "" {
```

File: aws/resource_aws_glue_security_configuration.go [hunk 2]
```diff
@@ -248,7 +250,9 @@ func expandGlueJobBookmarksEncryption(l []interface{}) *glue.JobBookmarksEncrypt
...
```
Comment (DrFaust92) [inline aws/resource_aws_glue_security_configuration.go:252]: same as above

**Review by DrFaust92 — CHANGES_REQUESTED** [review]
Minor comment, lets just run the basic test and ill verify all other. Looks good to me.

File: aws/resource_aws_glue_security_configuration.go (hunk 1 above)
Comment (AndresCidoncha) [reply aws/resource_aws_glue_security_configuration.go]: Done

File: aws/resource_aws_glue_security_configuration.go (hunk 2 above)
Comment (AndresCidoncha) [reply aws/resource_aws_glue_security_configuration.go]: Done

**Review by DrFaust92 — APPROVED** [review]
LGTM

```
--- PASS: TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSES3 (46.12s)
--- PASS: TestAccAWSGlueSecurityConfiguration_Basic (46.38s)
--- PASS: TestAccAWSGlueSecurityConfiguration_CloudWatchEncryption_CloudWatchEncryptionMode_SSEKMS (61.96s)
--- PASS: TestAccAWSGlueSecurityConfiguration_JobBookmarksEncryption_JobBookmarksEncryptionMode_CSEKMS (61.98s)
--- PASS: TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSEKMS (63.91s)
```

**Comment by breathingdust** [comment]
LGTM :rocket: Thanks @AndresCidoncha!

Verified Acceptance Tests in Commercial (us-west-2)

```console
make testacc TEST=./aws TESTARGS='-run=TestAccAWSGlueSecurityConfiguration_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSGlueSecurityConfiguration_ -timeout 120m
=== RUN   TestAccAWSGlueSecurityConfiguration_Basic
=== PAUSE TestAccAWSGlueSecurityConfiguration_Basic
=== RUN   TestAccAWSGlueSecurityConfiguration_CloudWatchEncryption_CloudWatchEncryptionMode_SSEKMS
=== PAUSE TestAccAWSGlueSecurityConfiguration_CloudWatchEncryption_CloudWatchEncryptionMode_SSEKMS
=== RUN   TestAccAWSGlueSecurityConfiguration_JobBookmarksEncryption_JobBookmarksEncryptionMode_CSEKMS
=== PAUSE TestAccAWSGlueSecurityConfiguration_JobBookmarksEncryption_JobBookmarksEncryptionMode_CSEKMS
=== RUN   TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSEKMS
=== PAUSE TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSEKMS
=== RUN   TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSES3
=== PAUSE TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSES3
=== CONT  TestAccAWSGlueSecurityConfiguration_Basic
=== CONT  TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSEKMS
=== CONT  TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSES3
=== CONT  TestAccAWSGlueSecurityConfiguration_JobBookmarksEncryption_JobBookmarksEncryptionMode_CSEKMS
=== CONT  TestAccAWSGlueSecurityConfiguration_CloudWatchEncryption_CloudWatchEncryptionMode_SSEKMS
--- PASS: TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSES3 (18.15s)
--- PASS: TestAccAWSGlueSecurityConfiguration_Basic (18.20s)
--- PASS: TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSEKMS (19.85s)
--- PASS: TestAccAWSGlueSecurityConfiguration_CloudWatchEncryption_CloudWatchEncryptionMode_SSEKMS (20.19s)
--- PASS: TestAccAWSGlueSecurityConfiguration_JobBookmarksEncryption_JobBookmarksEncryptionMode_CSEKMS (20.21s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	21.582s
```

Verified Acceptance Tests in GovCloud (us-gov-west-1)

```console
make testacc TEST=./aws TESTARGS='-run=TestAccAWSGlueSecurityConfiguration_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSGlueSecurityConfiguration_ -timeout 120m
=== RUN   TestAccAWSGlueSecurityConfiguration_Basic
=== PAUSE TestAccAWSGlueSecurityConfiguration_Basic
=== RUN   TestAccAWSGlueSecurityConfiguration_CloudWatchEncryption_CloudWatchEncryptionMode_SSEKMS
=== PAUSE TestAccAWSGlueSecurityConfiguration_CloudWatchEncryption_CloudWatchEncryptionMode_SSEKMS
=== RUN   TestAccAWSGlueSecurityConfiguration_JobBookmarksEncryption_JobBookmarksEncryptionMode_CSEKMS
=== PAUSE TestAccAWSGlueSecurityConfiguration_JobBookmarksEncryption_JobBookmarksEncryptionMode_CSEKMS
=== RUN   TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSEKMS
=== PAUSE TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSEKMS
=== RUN   TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSES3
=== PAUSE TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSES3
=== CONT  TestAccAWSGlueSecurityConfiguration_Basic
=== CONT  TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSEKMS
=== CONT  TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSES3
=== CONT  TestAccAWSGlueSecurityConfiguration_CloudWatchEncryption_CloudWatchEncryptionMode_SSEKMS
=== CONT  TestAccAWSGlueSecurityConfiguration_JobBookmarksEncryption_JobBookmarksEncryptionMode_CSEKMS
--- PASS: TestAccAWSGlueSecurityConfiguration_Basic (13.44s)
--- PASS: TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSES3 (13.49s)
--- PASS: TestAccAWSGlueSecurityConfiguration_JobBookmarksEncryption_JobBookmarksEncryptionMode_CSEKMS (15.47s)
--- PASS: TestAccAWSGlueSecurityConfiguration_S3Encryption_S3EncryptionMode_SSEKMS (15.68s)
--- PASS: TestAccAWSGlueSecurityConfiguration_CloudWatchEncryption_CloudWatchEncryptionMode_SSEKMS (15.80s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	17.126s
```

**Pull request merged**